### ISSUE TITLE
Plot diagnostics

### ISSUE THREAD

**Issue by dustinvtran** [airoldilab/sgd#36]
These must allow one to specify multiple sgd objects to plot.
- [x] MSE
- [x] Classification error
- [ ] Evaluation of cost function

available x-axis for each of the above plots:
- [x] Runtime
- [x] log-Iteration

**Comment by dustinvtran** [comment]
http://arxiv.org/abs/1505.02417 some examples of plot diagnostics we'd like (and also experiments to run)

**Comment by dustinvtran** [comment]
Store 100 uniformly separated parameter estimates as default, in log space

**Comment by dustinvtran** [comment]
For some idea of what the visuals look like, I quite like the stuff in http://arxiv.org/abs/1206.1106 for example (not particularly, just an arbitrary paper I chose). i.e. high resolution fonts, embedded LaTeX in the legend, etc.

**Comment by dustinvtran** [comment]
picture of current progress
![screen shot 2015-05-31 at 11 18 23 am](https://cloud.githubusercontent.com/assets/2569867/7902404/cd740b5c-0786-11e5-9478-7e3a9b3e48ff.png)
bugs/things to continue working on:
* sgd gives nonsensical prediction results (could be a result of bad learning rate)
* adagrad takes a disproportionately long time; not sure if this is just because it's O(d)
* sgd and adagrad give nonsensical cost function evaluations

**Comment by dustinvtran** [comment]
Progress:
![screen shot 2015-05-31 at 12 18 00 pm](https://cloud.githubusercontent.com/assets/2569867/7903836/b2455acc-07b6-11e5-8290-3200d2fa04d7.png)
1. was definitely just a problem of setting the hyperparameter `alpha` in the Xu's learning rate. This also still needs to be tweaked for implicit and ai-SGD
2. adagrad's bug, if it exists, lies in the C++ code. The time it takes is definitely proportional to the number of features. Covertype has 54 features, MNIST has 784, and Covertype is much faster as seen above
3. I still need to find the bug in this one
